### PR TITLE
Binary search tree

### PR DESCRIPTION
**Pull request by shakalee14**

### CODE REVIEW

File: src/binarySearchTree.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
+'use strict'
+
+function Node(value){
+  this.value = value
+  this.left = null
+  this.right = null
+  this.parent = null
+  this.leftOrRight = "root"
```
Comment (devonwesley) [inline src/binarySearchTree.js:8]: What's purpose does that "leftOrRight" value serve??

File: src/binarySearchTree.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    const recursion = (node) => {
+      if(node.value === value){
+        returnValue = node
+        return
+      }
+
+      if(node.left){
+        recursion(node.left)
+        if(returnValue !== null) {
+          return
```
Comment (devonwesley) [inline src/binarySearchTree.js:33]: Return?? what are we returning a ghost?

Comment (shakalee14) [reply src/binarySearchTree.js:33]: Exits the function almost like breaking

File: src/binarySearchTree.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+      if(node.right){
+        recursion(node.right)
+      }
+    }
+    recursion(this.root)
+    return returnValue
+  }
+
+  insert(value){
```
Comment (devonwesley) [inline src/binarySearchTree.js:45]: Nice! goodjob

File: src/binarySearchTree.js (hunk 2 above)
Comment (devonwesley) [reply src/binarySearchTree.js:33]: Okay if it work's. but if you want we can sit down and find a better way to think about this.

**Pull request merged**